src/reflect/scala/reflect/internal/Definitions.scala

```diff
@@ -1784,6 +1784,17 @@ trait Definitions extends api.StandardDefinitions {
       lazy val arrayClassMethod       = getMemberMethod(ScalaRunTimeModule, nme.arrayClass)
       lazy val wrapVarargsRefArrayMethod = getMemberMethod(ScalaRunTimeModule, nme.wrapRefArray)
       lazy val genericWrapVarargsRefArrayMethod = getMemberMethod(ScalaRunTimeModule, nme.genericWrapArray)
+      lazy val primitiveWrapArrayMethod = Seq(
+        getMemberMethod(ScalaRunTimeModule, nme.wrapBooleanArray),
+        getMemberMethod(ScalaRunTimeModule, nme.wrapByteArray),
+        getMemberMethod(ScalaRunTimeModule, nme.wrapCharArray),
+        getMemberMethod(ScalaRunTimeModule, nme.wrapIntArray),
+        getMemberMethod(ScalaRunTimeModule, nme.wrapDoubleArray),
+        getMemberMethod(ScalaRunTimeModule, nme.wrapFloatArray),
+        getMemberMethod(ScalaRunTimeModule, nme.wrapLongArray),
+        getMemberMethod(ScalaRunTimeModule, nme.wrapShortArray)
```

Comment:
It seems to also work for `Array[Unit](())` - could you add that to the test?

```suggestion
        getMemberMethod(ScalaRunTimeModule, nme.wrapShortArray),
        getMemberMethod(ScalaRunTimeModule, nme.wrapUnitArray)
```

Reply:
Test will appear these method calls, do not know whether it is right?
```
Method call statistics:
    1  scala/runtime/BoxedUnit.<clinit>()V
    1  scala/runtime/BoxedUnit.<init>()V

```

Reply:
That's OK; `Array[Unit]` is `Array[BoxedUnit]` at runtime, the instance needs to be crated.

Reply:
```
1. It seems to be optimized
  #26 = NameAndType        #24:#25        // "<init>":()V
  #27 = Methodref          #4.#26         // java/lang/Object."<init>":()V
  #28 = Utf8               scala/runtime/BoxedUnit
  #29 = Class              #28            // scala/runtime/BoxedUnit
  #30 = Utf8               UNIT
  #31 = Utf8               Lscala/runtime/BoxedUnit;
  #32 = NameAndType        #30:#31        // UNIT:Lscala/runtime/BoxedUnit;
  #33 = Fieldref           #29.#32        // scala/runtime/BoxedUnit.UNIT:Lscala/runtime/BoxedUnit;


2.
  #65 = Utf8               scala/runtime/BoxedUnit
  #66 = Class              #65            // scala/runtime/BoxedUnit
  #67 = Utf8               UNIT
  #68 = Utf8               Lscala/runtime/BoxedUnit;
  #69 = NameAndType        #67:#68        // UNIT:Lscala/runtime/BoxedUnit;
  #70 = Fieldref           #66.#69        // scala/runtime/BoxedUnit.UNIT:Lscala/runtime/BoxedUnit;
  #71 = Utf8               wrapUnitArray
  #72 = Utf8               ([Lscala/runtime/BoxedUnit;)Lscala/collection/immutable/ArraySeq;
  #73 = NameAndType        #71:#72        // wrapUnitArray:([Lscala/runtime/BoxedUnit;)Lscala/collection/immutable/ArraySeq;
  #74 = Methodref          #43.#73        // scala/runtime/ScalaRunTime$.wrapUnitArray:([Lscala/runtime/BoxedUnit;)Lscala/collection/immutable/ArraySeq;
  #75 = Utf8               Unit
  #76 = Utf8               ()Lscala/reflect/ManifestFactory$UnitManifest;
  #77 = NameAndType        #75:#76        // Unit:()Lscala/reflect/ManifestFactory$UnitManifest;
  #78 = Methodref          #52.#77        // scala/reflect/ClassTag$.Unit:()Lscala/reflect/ManifestFactory$UnitManifest;
  #79 = Class              #21            // "[Lscala/runtime/BoxedUnit;"
```